aderyn_core/src/framework/foundry.rs

```diff
@@ -113,23 +98,85 @@ pub fn load_foundry(foundry_root: &PathBuf) -> Result<LoadedFoundry, Box<dyn Err
     let output_filepaths = get_matching_output_files(&json_output_filepaths, &contract_filepaths);
 
     Ok(LoadedFoundry {
-        src_path: foundry_config.profile.default.src,
+        src_path: foundry_config.src,
         src_filepaths: contract_filepaths,
         output_filepaths,
     })
 }
 
 fn read_config(path: &PathBuf) -> Result<FoundryConfig, Box<dyn Error>> {
     let contents = read_to_string(path).unwrap();
-    let foundry_config_toml = toml::from_str(&contents);
+
+    let foundry_config_toml = contents.parse::<Table>();
+
     let foundry_config = match foundry_config_toml {
         Ok(config) => config,
         Err(e) => {
             eprintln!("Error parsing TOML: {:?}", e);
             std::process::exit(1);
         }
     };
-    Ok(foundry_config)
+
+    let foundry_profile = std::env::var("FOUNDRY_PROFILE")
+        .unwrap_or("default".to_string())
+        .to_lowercase();
+
+    if let Some(foundry_profiles) = foundry_config["profile"].as_table() {
+        // println!("{:#?}", foundry_profiles);
+
+        let default_profile = foundry_profiles.get("default");
+
+        for (profile_key, value) in foundry_profiles {
+            if profile_key.contains(&foundry_profile) {
+                if let Some(value) = value.as_table() {
+                    let profile_src = {
+                        if let Some(toml::Value::String(src)) = value.get("src") {
+                            src.clone()
+                        } else if let Some(default_profile) = default_profile {
+                            if let Some(toml::Value::String(src)) = default_profile.get("src") {
```

Comment:
If `src` is not found in the `foundry.toml` section of the relevant profile, we fallback to the default profile

Similarly for `out`